pandas/core/generic.py

```diff
@@ -10358,6 +10358,39 @@ def _logical_func(
             filter_type="bool",
         )
 
+    @overload
+    def any(
+        self: DataFrame,
+        axis: Axis = ...,
+        bool_only: bool_t | None = ...,
+        skipna: bool_t = ...,
+        level: Level | None = ...,
+        **kwargs,
+    ) -> Series:
+        ...
+
+    @overload
+    def any(
+        self: Series,
+        axis: Axis = ...,
+        bool_only: bool_t | None = ...,
+        skipna: bool_t = ...,
+        level: Level | None = ...,
+        **kwargs,
+    ) -> bool_t:
+        ...
+
+    @overload
+    def any(
+        self: NDFrame,
```

Comment:
Sorry, I don't know the solution. I think you don't need the `NDFrame` annotation for `self` (that will not help with the error).

I opened https://github.com/python/mypy/issues/11759 to see whether this is a false-positive in mypy.

Reply:
Thanks for the help @twoertwein